swift-paperless/Views/TLSListView.swift

```diff
@@ -0,0 +1,231 @@
+                return
+            }
+            if validateCertificate(certificateData: data, certificatePassword: certificatePassword) {
+                certificateState = .valid
+                isCertificateValid = true
+            } else {
+                certificateState = .wrongPassword
+                isCertificateValid = false
+            }
+        }
+
+        var body: some View {
+            Form {
+                Section {
+                    TextField(String(localized: .localizable(.name)), text: $certificateName)
+                    SecureField(String(localized: .login(.password)), text: $certificatePassword).autocorrectionDisabled()
+                    Button(String(localized: .settings(.selectCertificate))) {
+                        isImporting = true
+                    }
+                } footer: {
+                    switch certificateState {
+                    case .notloaded:
+                        Text(String(localized: .settings(.certificateNotLoaded)))
+                    case .wrongPassword:
+                        Text(String(localized: .settings(.certificateLoadError)))
+                    case .valid:
+                        HStack {
+                            Image("checkmark.circle.fill")
+                        }
+                    }
+                }
+
+                Section {
+                    Button(String(localized: .localizable(.save))) {
+                        guard let data = certificateData else {
+                            isCertificateValid = false
+                            certificateState = .notloaded
+                            return
+                        }
+                        saveToKeychain(certificateData: data, certificatePassword: certificatePassword, certificateName: certificateName)
+                        dismiss()
+                    }.disabled(!isCertificateValid)
+                }
+            }
+            .onChange(of: certificateData) {
+                validateInput()
+            }
+            .onChange(of: certificatePassword) {
+                validateInput()
+            }
+            .fileImporter(
+                isPresented: $isImporting,
+                allowedContentTypes: [.pkcs12],
+                allowsMultipleSelection: false
+            ) { result in
+                do {
+                    guard let selectedFile: URL = try result.get().first else { return }
+                    if selectedFile.startAccessingSecurityScopedResource() {
+                        certificateData = try Data(contentsOf: selectedFile)
+                        defer { selectedFile.stopAccessingSecurityScopedResource() }
+                        certificateName = selectedFile.lastPathComponent
+                    } else {
+                        Logger.shared.error("Error opening File from secure context")
+                    }
+                } catch {
+                    Logger.shared.error("erro while loding PfX \(error)")
```

Comment:
```suggestion
                    Logger.shared.error("Error while loding PfX \(error)")
```

Consider surfacing these errors.